src/omero_zarr/masks.py

```diff
@@ -23,53 +39,199 @@ def image_masks_to_zarr(image, args):
 
         if len(mask_shapes) > 0:
             masks[roi.id.val] = mask_shapes
+            shape_count += len(mask_shapes)
 
-    print(f"Found {len(masks)} masks")
+    print(f"Found {shape_count} mask shapes in {len(masks)} ROIs")
 
-    if masks:
-        stack = masks_to_zarr(masks, image)
-        name = f"{image.id}_masks.zarr"
-        root = zarr.open_group(name, mode="w")
-        za = root.create(
-            "0",
-            shape=stack.shape,
-            chunks=(1, 1, size_y, size_x),
-            dtype=stack.dtype,
-        )
+    dtype = MASK_DTYPE_SIZE[int(args.mask_bits)]
 
-        za[:, :, :, :] = stack
-
-        print("Created", name)
+    if masks:
+        if args.group_rois:
+            for (roi_id, roi) in masks.items():
+                _save_masks([roi], image, str(roi_id), dtype)
```

Comment:
Another idea beyond separate arrays versus a labelled image would be a 6D array, right?

Reply:
Yes.

Reply:
I'll open this in a follow-up PR.